lib/src/over_react_test/prop_type_utils.dart

```diff
@@ -0,0 +1,87 @@
+import 'dart:html';
+import 'dart:js';
+
+import 'package:matcher/matcher.dart';
+import 'package:meta/meta.dart';
+import 'package:over_react/component_base.dart';
+import 'package:over_react_test/jacket.dart';
+import 'package:react/react_client/react_interop.dart' show PropTypes;
+import 'package:test/test.dart';
+
+/// Tests the prop types of a component by calling the props, adding children if necessary, and checking the console
+/// for expected log output.
+void testPropTypesWithUiProps(
+    {@required UiProps componentProps, dynamic childProps, String customErrorMessage, Element mountNode}) {
```

Comment:
Instead of having 2 helpers, maybe we could just have one and add a named arg here that is called like `willThrow` or `doesThrow` that is a `bool` along with the `errorMatcher` arg and wrap the mount in the function call and just like in the expect be like `doesThrow ? errorMatcher : returnsNormally` and then maybe just default error matcher to `throwsA(anything)`

Reply:
I change my mind, just have a `errorMatcher` arg without a `willThrown` and just do a null check on that, `errorMatcher != null ? errorMatcher : returnsNormally`